scripts/dll/build-dlls.js

```diff
@@ -7,53 +7,109 @@
 
-const ROOT_DIRECTORY = path.resolve( __dirname, '..', '..' );
-const IS_DEVELOPMENT_MODE = process.argv.includes( '--dev' );
+const argv = minimist( process.argv.slice( 2 ) );
+
+const ROOT_DIRECTORY = argv.cwd ? path.resolve( argv.cwd ) : path.resolve( __dirname, '..', '..' );
+const BASE_DLL_CONFIG_PATH = argv[ 'base-dll-config' ] ? path.resolve( argv[ 'base-dll-config' ] ) : null;
+const BASE_DLL_PATH = argv[ 'base-dll-path' ];
+const IS_DEVELOPMENT_MODE = argv.dev;
+const VERBOSE_MODE = argv.verbose;
+
+// Lets highlight and space out the messages in the verbose mode to
+// make them stand out from the wall of text that webpack spits out.
+const prefix = VERBOSE_MODE ? '\n📍 ' : '';
+
+if ( BASE_DLL_CONFIG_PATH ) {
+	console.log( prefix + chalk.bold( 'Creating the base DLL build...' ) );
 
-if ( IS_DEVELOPMENT_MODE ) {
-	console.log( '🛠️️  ' + chalk.yellow( 'Development mode is active.' ) );
-} else {
-	console.log( '⚠️  ' + chalk.magenta( 'Production mode is active. Use --dev to build in the development mode.' ) );
+	execute( {
+		command: [ 'webpack', `--config=${ BASE_DLL_CONFIG_PATH }` ],
+		cwd: BASE_DLL_PATH || ROOT_DIRECTORY
+	} );
 }
 
-// -------------------------------------------------------------
-// ------------------------------------------- Base DLL build --
+console.log( prefix + chalk.bold( 'Creating DLL-compatible package builds...' ) );
 
-console.log( '\n📍 ' + chalk.cyan.underline( 'Creating the base DLL build...\n' ) );
+getPackageNames( ROOT_DIRECTORY )
+	.filter( isNotBaseDll )
+	.filter( hasDLLBuildScript )
+	.forEach( fullPackageName => {
+		console.log( prefix + `Building ${ fullPackageName }...` );
 
-const webpackArguments = [ '--config=./scripts/dll/webpack.config.dll.js' ];
+		execute( {
+			command: [ 'yarn', 'run', 'dll:build' ],
+			cwd: path.join( ROOT_DIRECTORY, 'packages', fullPackageName )
+		} );
+	} );
 
-if ( IS_DEVELOPMENT_MODE ) {
-	webpackArguments.push( '--mode=development' );
+/**
+ * @param {String} cwd
+ * @returns {Array<String>}
+ */
+function getPackageNames( cwd ) {
+	return childProcess.execSync( 'ls -1 packages', {
+		encoding: 'utf8',
+		cwd
+	} ).toString().trim().split( '\n' );
 }
 
-childProcess.spawnSync( 'webpack', webpackArguments, {
-	encoding: 'utf8',
-	shell: true,
-	cwd: ROOT_DIRECTORY,
-	stdio: 'inherit',
-	stderr: 'inherit'
-} );
+/**
+ * @param {String} name
+ * @returns {Boolean}
+ */
+function isNotBaseDll( name ) {
+	return name !== 'ckeditor5-dll';
+}
 
-// -------------------------------------------------------------
-// ---------------------------- DLL-compatible package builds --
+/**
+ * @param {String} name
+ * @returns {Boolean}
+ */
+function hasDLLBuildScript( name ) {
+	const packageJsonPath = path.join( process.cwd(), 'packages', name, 'package.json' );
+	const scripts = require( packageJsonPath ).scripts;
 
-console.log( '\n📍 ' + chalk.underline( 'Creating DLL-compatible package builds...\n' ) );
+	return scripts && scripts[ 'dll:build' ];
```

Comment:
```suggestion
	return !!( scripts && scripts[ 'dll:build' ] );
```

To match the JSDoc signature.